configs/configs_test.go

```diff
@@ -16,30 +16,37 @@ package configs
 
 import (
 	"encoding/json"
+	"fmt"
 	"testing"
 	"time"
 )
 
 func TestJSONDuration(t *testing.T) {
-	var s struct {
-		Dur JSONDuration
+	testCases := []struct {
+		json         []byte
+		wantErr      string
+		wantDuration time.Duration
+	}{
+		{[]byte(`{"dur":"10s"}`), "", time.Second * 10},
+		{[]byte(`{"dur": 1}`), "json: cannot unmarshal number into Go struct field .Dur of type string", 0},
+		{[]byte(`{"dur": ""}`), "time: invalid duration ", 0},
 	}
-
-	a := []byte(`{"dur":"10s"}`)
-	if err := json.Unmarshal(a, &s); err != nil {
-		t.Errorf("Failed to unmarshal duration: %v", err)
-	}
-	if s.Dur.Duration != (time.Second * 10) {
-		t.Errorf("Unexpected duration, want: 10s, got: %s", s.Dur)
-	}
-
-	b := []byte(`{"dur": 1}`)
-	if err := json.Unmarshal(b, &s); err == nil {
-		t.Error("Didn't fail when unmarshalling a non-string type into JSONDuration")
-	}
-
-	c := []byte(`{"dur": ""}`)
-	if err := json.Unmarshal(c, &s); err == nil {
-		t.Error("Didn't fail when unmarshalling an empty string into JSONDuration")
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("json: %s", tc.json), func(t *testing.T) {
+			var s struct {
+				Dur JSONDuration
+			}
+			err := json.Unmarshal(tc.json, &s)
```

Comment:
Suggestion:

```go
err := json.Unmarshal(tc.json, &s)
switch hasErr, wantErr := err != nil, tc.wantErr != ""; {
case hasErr != wantErr:
  t.Errorf(...)
case hasErr && err.Error() != tc.wantErr:
  t.Errorf(...)
}
```

Although, if you take the comment above about the error string, the case comparing the messages goes away, so we could simplify it further to an if.